### PR TITLE
core: add unit tests for arrival on stop signal for train schedule

### LINKED ISSUE #7044: Create a test generating spacing conflicts
## Description

Create a test that should create a conflict with the current implementation. This test should not create a conflict anymore once "stop on closed-signal" has been implemented.

### Tests on all dimensions 

* tests on all dimensions : near/far trains, open/closed signal, before/after seeing signal, first/second train: conflict disappears for second near trains stopping on closed-signal, or if trains stops before seeing any protecting signal.
* ~maybe also both trains stopping~

### ~Manual demo~

* ~manual "demo" test using frontend to trigger a conflict to be resolved when stop on closed-signal will be implemented~

### LINKED ISSUE #7045: Create test generating path (routing) conflicts
## Description and goal 

Create a test that should create a conflict with the current implementation. This test should not create a conflict anymore once "stop on closed-signal" has been implemented.

### Tests on all dimensions 

* tests on all dimensions : near/far trains, open/closed signal, before/after seeing signal, first/second train: conflict disappears for second near trains stopping on closed-signal, or if trains stops before seeing any protecting signal.
* ~maybe also both trains stopping~

### ~Manual demo~

* ~manual "demo" test using frontend to trigger a conflict to be resolved when stop on closed-signal will be implemented~

### PR DESCRIPTION
**Pull request by bougue-pe**
Test more combinations and "corner" cases on spacing and routing conflicts.

Fix: the routing requirement processing is now waiting to see a signal before requiring block, and consistent with spacing requirement handling.

closes https://github.com/OpenRailAssociation/osrd/issues/7044
closes https://github.com/OpenRailAssociation/osrd/issues/7045

### CODE REVIEW

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java [hunk 1]
```diff
@@ -250,14 +252,185 @@ In practice, the stop train will require the route in front of him (zone and swi
...
+            var reqA = convertRequirements(0L, 0.0, simResultAWithStop.train);
+            var reqB = convertRequirements(1L, 0.0, simResultB.train);
+            var conflicts = ConflictsKt.detectConflicts(List.of(reqA, reqB));
             assertFalse(conflicts.stream().anyMatch((conflict) -> conflict.conflictType == ROUTING));
             assertFalse(conflicts.stream().anyMatch((conflict) -> conflict.conflictType == SPACING));
         }
     }
 
+    /*
+    Test for train starting after the first signal protecting block with potential conflict.
```
Comment (woshilapin) [inline core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: The sentence is long and it's not obvious to which part of the sentence refers "with potential conflicts" (I guess it's the train?).

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java [hunk 2]
```diff
@@ -250,14 +252,185 @@ In practice, the stop train will require the route in front of him (zone and swi
...
+    Then the train stops either (all combinations):
+    * on stop-signal or not
```
Comment (woshilapin) [inline core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: Asked myself if it was "stop or not" vs "stop-signal or open signal". After reading again, the structure of the sentence is correct to understand without a doubt, but I would rather make it more explicit anyway.
```suggestion
    * on stop-signal or open signal
```

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java [hunk 3]
```diff
@@ -250,14 +252,185 @@ In practice, the stop train will require the route in front of him (zone and swi
...
+                simpleSim(fullInfra, pathPropsNorth, chunkPathNorth, 0, Double.POSITIVE_INFINITY, List.of());
+
+        var reqWithStop = convertRequirements(0L, 0.0, simResultCenterWithStop.train);
+        var reqOvertaking = convertRequirements(1L, directStartTime, simResultNorthOvertaking.train);
+        var conflicts = ConflictsKt.detectConflicts(List.of(reqWithStop, reqOvertaking));
+        assert (hasRoutingConflict == conflicts.stream().anyMatch((conflict) -> conflict.conflictType == ROUTING));
+        assert (hasSpacingConflict == conflicts.stream().anyMatch((conflict) -> conflict.conflictType == SPACING));
+    }
+
+    @ParameterizedTest
```
Comment (woshilapin) [inline core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: This test seems very similar to the one above the above. First, it'd be nice to put them together. Then, a comment like for the others would help.

**Review by woshilapin — APPROVED** [review]
So many tests 🎩Mostly some minor improvements about comments.

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java (hunk 2 above)
Comment (bougue-pe) [reply core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: Done in https://github.com/OpenRailAssociation/osrd/pull/7704/commits/66ce90c73d8856e46d02fb8522da75d4a0b5bb54

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java (hunk 1 above)
Comment (bougue-pe) [reply core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: It was the block.
I added details to clarify it in https://github.com/OpenRailAssociation/osrd/pull/7704/commits/66ce90c73d8856e46d02fb8522da75d4a0b5bb54
Hope it helps.

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java (hunk 3 above)
Comment (bougue-pe) [reply core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: Moved the test and commented in https://github.com/OpenRailAssociation/osrd/pull/7704/commits/864991bab8e6de95f0b4b2231e4fadec85098d44

**Review by woshilapin — APPROVED** [review]
Thanks for the fix on the comments.

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java [hunk 4]
```diff
@@ -250,14 +252,196 @@ In practice, the stop train will require the route in front of him (zone and swi
...
+    Also, the stopping train stops either (all combinations):
+    * on stop-signal or on open signal
+    * while seeing the last signal protecting block or before seeing it
+
+    Stopping on open signal and before seeing any signal protecting the block is quite an undefined behavior (train
+    is starting after first signal).
+    In current implementation, the considered signal is not really the limiting one, but the first signal seen by the
+    train that is protecting the block.
+    So one might consider that train should require resource at its very beginning, but we actually go for waiting
+    to see any signal.
```
Comment (Khoyo) [inline core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: That's specifing to routing conflicts ;)

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java [hunk 5]
```diff
@@ -250,14 +252,196 @@ In practice, the stop train will require the route in front of him (zone and swi
...
+    @ParameterizedTest
+    @CsvSource({
+        "false, 100, true, true",
+        "true, 100, false, false",
+        "false, -100, false, false",
+        "true, -100, false, false",
+    })
+    public void conflictHandlingWithTrainStartingDuringRequirementTrigger(
+            boolean onStopSignal,
+            double distanceAfterLastSignalSight,
```
Comment (Khoyo) [inline core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: It's unclear what this parameter does from it's name. I'd rather have a boolean `seeingEntrySignal` or something like that.

**Review by Khoyo — APPROVED** [review]
Otherwise LGTM

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java (hunk 4 above)
Comment (bougue-pe) [reply core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: Added a little precision on comment https://github.com/OpenRailAssociation/osrd/pull/7704/commits/3d698711d5ce6cfb2c4c57b9039399fd914fb8ef
As mentioned aside, not sure how it works for spacing case, so I didn't comment on this.

File: core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java (hunk 5 above)
Comment (bougue-pe) [reply core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java]: Switch done in https://github.com/OpenRailAssociation/osrd/pull/7704/commits/893a440284a753f05b51ecca1b87991625dd6ec8 (same for limitingSignal in https://github.com/OpenRailAssociation/osrd/pull/7704/commits/db81e2e0fcc727ddc13b248276126ce15731b326).

**Pull request merged**